### ISSUE TITLE
Next.js 13 - Builds not outputting css on Windows

### ISSUE THREAD

**Issue by exneval** [vanilla-extract-css/vanilla-extract#1086]
### Describe the bug

- Clone the ["should be working repo"](https://github.com/SuttonJack/vanilla-extract-app-dir) in Windows

- Inspect with DevTools, and see footer class being generated without styles

![image](https://github.com/vanilla-extract-css/vanilla-extract/assets/3431785/f837980e-8d8b-450b-8e0f-2dc0a8fdd1d1)

Need help @markdalgleish @SuttonJack 

[Reference in discussion](https://github.com/vanilla-extract-css/vanilla-extract/issues/929#issuecomment-1539274116)

Edit:
Running the app inside container seems fine.

### Reproduction

https://github.com/SuttonJack/vanilla-extract-app-dir

### System Info

```shell
Windows 11
```

### Used Package Manager

pnpm

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by MatheusLittig** [comment]
I'm having the same issue here on Windows 10

**Comment by askoufis** [comment]
I'm still in the process of setting up a windows dev environment, but I'll take a look at it when I can.

**Comment by i-bsd** [comment]
Also happening in Linux (Ubuntu).

**Comment by askoufis** [comment]
> Also happening in Linux (Ubuntu).

It works fine for me, on MacOS though, but I wouldn't expect it to behave very differently between ubuntu and macos.

Also general update, I'm reproduce the issue on windows. It's odd that classnames are in the markup, but there are no CSS files in the document `head`. This seems like an issue with Next, but it might not be. We've made a discussion in the next repo about Vanilla Extract support, pointing to various next bugs https://github.com/vercel/next.js/discussions/49892.

**Comment by mtripg6666tdr** [comment]
Same here on Windows 11, but not be reproduced on Ubuntu (WSL).

**Comment by NotNite** [comment]
Same here on Windows 10 (19045.3208), Node.js 19.2.0, latest versions of `@vanilla-extract/css`/`next`/`@vanilla-extract/next-plugin`.

**Comment by Sharlottes** [comment]
Same here.
reproduction: https://github.com/Sharlottes/ve-missing-style-reproduction
interestingly, the deployment on the Vercel works well.
https://ve-missing-style-reproduction.vercel.app/

**Comment by GeeWizWow** [comment]
still doing some debugging, but it's server components specifically that are failing, adding a `'use client'` directive results in stylesheets being generated.

**Comment by leljak** [comment]
> still doing some debugging, but it's server components specifically that are failing, adding a 'use client' directive results in stylesheets being generated.

This works for components, what about global styles, any workaround?

**Comment by GeeWizWow** [comment]
> > still doing some debugging, but it's server components specifically that are failing, adding a 'use client' directive results in stylesheets being generated.
> 
> This works for components, what about global styles, any workaround?

I did find a workaround, but ended not needing it, as it's a bit of an _inelegant hack_, but if you use a wildcard import and ensure you use the resulting variable in some capacity, it should pickup the globalStyles

```
import * as styles from './global.css';

console.log(styles)
```

**Comment by leljak** [comment]
> > > still doing some debugging, but it's server components specifically that are failing, adding a 'use client' directive results in stylesheets being generated.
> > 
> > 
> > This works for components, what about global styles, any workaround?
> 
> I did find a workaround, but ended not needing it, as it's a bit of an _inelegant hack_, but if you use a wildcard import and ensure you use the resulting variable in some capacity, it should pickup the globalStyles
> 
> ```
> import * as styles from './global.css';
> 
> console.log(styles)
> ```

Uh, really ugly, but it works, thank you!

**Comment by syfxlin** [comment]
I've recently been migrating my blog from Gatsby to Next.js and I'm having the same problem. I found a solution at [style9-webpack](https://github.com/SukkaW/style9-webpack) and made another Next.js plugin for vanilla-extract that works fine on Windows/Linux/macOS, if you're having a similar problem, you can try [@syfxlin/next-plugin-vanilla-extract](https://github.com/syfxlin/next-plugin-vanilla-extract).

**Comment by GeeWizWow** [comment]
> I've recently been migrating my blog from Gatsby to Next.js and I'm having the same problem. I found a solution at [style9-webpack](https://github.com/SukkaW/style9-webpack) ([SukkaW/style9-webpack#1](https://github.com/SukkaW/style9-webpack/issues/1)) and made another Next.js plugin for vanilla-extract that works fine on Windows/Linux/macOS, if you're having a similar problem, you can try [@syfxlin/next-plugin-vanilla-extract](https://github.com/syfxlin/next-plugin-vanilla-extract).

Is this something that can be contributed here? I'm sure many of us would prefer not to see fragmentation, especially as vanilla-extract and or next continue to evolve

**Comment by syfxlin** [comment]
Hi @GeeWizWow, @syfxlin/next-plugin-vanilla-extract hasn't been fully tested, for example I'm not sure if it works in the Next.js Pages Directory yet. Once it's confirmed to be reliable, I can contribute here, thanks!

**Comment by sherluok** [comment]
I temporarily gave up on the `next-plugin` and wrote the webpack transformer myself with `webpack-plugin`.
It works now, just for reference, don not use it in production：

```js
// 2023/10/03; windows-11; next^13.5.3; webpack-plugin^2.3.1; css-loader^6.8.1

const { VanillaExtractPlugin } = require('@vanilla-extract/webpack-plugin/next');
const { default: MiniCssExtractPlugin } = require('next/dist/build/webpack/plugins/mini-css-extract-plugin');

/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    serverActions: true,
  },
  webpack(config) {
    config.plugins.unshift(new MiniCssExtractPlugin());
    config.plugins.unshift(new VanillaExtractPlugin());

    config.module.rules.unshift({
      test: /vanilla\.virtual\.css$/,
      use: [
        MiniCssExtractPlugin.loader,
        require.resolve('css-loader'),
        VanillaExtractPlugin.loader,
      ],
    });
    return config;
  },
};

module.exports = nextConfig;
```

**Comment by askoufis** [comment]
> I gave up `next-plugin` and instead write webpack transformer with `webpack-plugin` directly. It works for now, just for reference, do not use this in production:
> 
> ```js
> // 2023/10/03; windows 11; next@13.5.3; @vanilla-extract/webpack-plugin@2.3.1
> 
> const { VanillaExtractPlugin } = require('@vanilla-extract/webpack-plugin/next');
> const { default: MiniCssExtractPlugin } = require('next/dist/build/webpack/plugins/mini-css-extract-plugin');
> 
> /** @type {import('next').NextConfig} */
> const nextConfig = {
>   webpack(config) {
>     config.plugins.unshift(new MiniCssExtractPlugin());
>     config.plugins.unshift(new VanillaExtractPlugin());
> 
>     config.module.rules.unshift({
>       test: /vanilla\.virtual\.css$/,
>       use: [
>         MiniCssExtractPlugin.loader,
>         require.resolve('css-loader'),
>         VanillaExtractPlugin.loader,
>       ],
>     });
>     return config;
>   },
> };
> 
> module.exports = nextConfig;
> ```

Does the fix to the next plugin from https://github.com/vanilla-extract-css/vanilla-extract/pull/1180 work for you?

**Comment by plrs9816** [comment]
https://github.com/plrs9816/vanilla-extract-app-dir

Something broke again in next 13.5?

In 13.4 it works fine but the css is not generated in 13.5. (I only tested in windows)

Strangely the above demo is working, because I added unnecessary `appDir: true` config.
`⚠ App router is available by default now, `experimental.appDir` option can be safely removed.`
It says the option does nothing, but if it's removed it doesn't work.
13.4 works fine without the option.
```
const nextConfig = {
  experimental: {
    appDir: true,
  },
};
```
